hugo/layouts/shortcodes/youtube.html

```diff
@@ -0,0 +1,33 @@
+{{- $id := .Get "id" | default (.Get 0) -}}
+{{- $title := .Get "title" | default "" }}
+{{- $image := .Get "image" | default "" }}
+{{- $url := "https://www.youtube.com/watch?v={{ $id }}" }}
+{{- $contentClass := "video__content" }}
+{{- $lazy := cond (or (eq (.Get "lazy") false) (eq (.Get "lazy") "false")) false true -}}
+
+{{ if not $title }}
+    {{- $contentClass = print $contentClass " video__content--no-title" -}}
+{{ end }}
+
+<div class="video" data-youtube="{{ $id }}">
+    {{ if $image }}
+        <img class="video__image" src="{{ $image }}" alt="{{ $title }}"{{ if $lazy }} loading="lazy"{{ end }}>
+    {{ else }}
+        <img class="video__image" src="https://img.youtube.com/vi/{{ $id }}/maxresdefault.jpg" alt="{{ $title }}"{{ if $lazy }} loading="lazy"{{ end }}>
```

Comment:
double check if the youtube img url doesn't set cookies. Otherwise we can maybe use the no-cookie youtube domain but I'm not sure that's available with the img service

Reply:
Did some digging, nocookies url is not available for the images. 
These are the cookies that come up on the example page:
![Screenshot 2023-08-15 at 13 50 54](https://github.com/cue-lang/cuelang.org/assets/122522301/088b949b-dde3-4e9b-907c-2fdac497882c)

Reply:
Did some googling and found we might be able to use the youtube api v3 urls: 

`https://i1.ytimg.com/vi/<insert-youtube-video-id-here>/default.jpg -   default
https://i1.ytimg.com/vi/<insert-youtube-video-id-here>/mqdefault.jpg - medium 
https://i1.ytimg.com/vi/<insert-youtube-video-id-here>/hqdefault.jpg - high
https://i1.ytimg.com/vi/<insert-youtube-video-id-here>/sddefault.jpg - standard
https://i1.ytimg.com/vi/<insert-youtube-video-id-here>/maxresdefault.jpg - max`

these are not from the youtube domain and should be safe to use with privacy (gdpr) settings in mind

Reply:
Using these links and after removing + refreshing the page, I still see the VISITOR_INFO1_LIVE and YSC cookies coming up.

Reply:
hm VISITOR_INFO1_LIVE seems to be a tracking cookie:
https://cookiedatabase.org/cookie/youtube/visitor_info1_live/
https://policies.google.com/technologies/cookies/embedded?hl=en-US

Seems like we can't use the youtube image as a fallback because it's not GDPR compliant.
So probably we should make the image attribute on the shortcode required?